private/buf/buffetch/buffetch.go

```diff
@@ -157,8 +157,24 @@ type RefParser interface {
 // NewRefParser returns a new RefParser.
 //
 // This defaults to dir or module.
-func NewRefParser(logger *zap.Logger) RefParser {
-	return newRefParser(logger)
+func NewRefParser(logger *zap.Logger, options ...NewRefParserOption) RefParser {
+	return newRefParser(logger, options...)
+}
+
+// NewRefParserOption is an option for NewRefParser. NewRefParserOptions are only accepted
+// for the default RefParser constructor.
+type NewRefParserOption func(*newRefParserOption)
```

Comment:
For constructors, we generally attach the options directly to the struct type, i.e. `refParser`, so you'll put `allowSingleFileRef` (soon `allowProtoFileRef`) directly on `refParser`.